File: diofant/informatics/002_mods.c
#include<math.h>
#include<time.h>

int MAX=1000000;

void solve();

int main() {
	
	time_t before = time(0);
	solve();
	double diff = difftime(time(0), before);
	
	printf("Time is %f\n", diff);
	
	return 0;
}

void solve() {
	int current=1;
	int mods_count = 0;
	int prev_mods_count = 0;
	int i = 0;
	
	while(current<MAX) {
	
		mods_count = 0;
		int sqr = sqrt(current);
		for(i=1; i<sqr; i++) {
			
			if(current%i == 0)
				mods_count++;
		}
		
		if(mods_count > prev_mods_count) {
			printf("%8d Mods count: %4d\n", current, mods_count);
			prev_mods_count = mods_count;
		}
		
		current++;
	}
}

File: diofant/informatics/002_mods.rb
# Чему равно наименьшее натуральное число меньшее 1 миллиона, которое имеет максимальное количество различных делителей.

class ModsSolver
  
  def initialize(max=1000000)
    @MAX = max
  end
  
  def solve_by_search_all

	current = 1
	mods_num = 0
	prev_mods_num = 0
	
	while current < @MAX
	  
	  mods_num = 0
	  sqr = Math.sqrt(current)
	  1.upto(sqr) do |md|
		if current % md == 0
		  mods_num += 1
		end
	  end

	  if mods_num > prev_mods_num
		printf "%6.0f: Mods count is %3.0f\n", current, mods_num
		prev_mods_num = mods_num
	  end

	  current += 1
	end
	
  end
  
end

def time
  start = Time.now
  yield
  Time.now - start
end

if __FILE__ == $0

  ms = ModsSolver.new
  
  takes = time do
	ms.solve_by_search_all
  end
  
  puts takes
end
